elifepubmed/generate.py

```diff
@@ -603,17 +613,11 @@ def set_grants(parent, poa_article):
 
 def dataset_assigning_authority(assigning_authority, uri):
     """precise assigning_authority value considering the uri in some cases"""
-    authority_map = OrderedDict([('NCBI', [])])
-    # tuple in form: uri hint, new assigning authority
-    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/geo', 'NCBI:geo'))
-    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/projects/gap', 'NCBI:dbgap'))
-    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/nuccore', 'NCBI:nucleotide'))
-    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/sra', 'NCBI:sra'))
-
-    if assigning_authority in authority_map:
-        for hint, new_value in authority_map.get(assigning_authority):
+    if ASSIGNING_AUTHORITY_MAP and assigning_authority in ASSIGNING_AUTHORITY_MAP:
```

Comment:
I don't think you need this line. If `ASSIGNING_AUTHORITY_MAP.get(assigning_authority)` does not return a value the for loop will be skipped.

Reply:
I guess so. I attempted to remove it, however it pushes the problem further up is `assigning_authority` is `None` then the `.get` fails. I will be expressing it differently though in the end.

Reply:
Hmm... I've tried it for a bit and I like the original better. `ASSIGNING_AUTHORITY_MAP.get(None)` isn't iterable, and neither is `ASSIGNING_AUTHORITY_MAP.get('other')`, both examples for tests that were passing with the if statement as I had it.

Reply:
I believe `ASSIGNING_AUTHORITY_MAP.get(assigning_authority, [])` would have given you what you were after.

Reply:
> I believe `ASSIGNING_AUTHORITY_MAP.get(assigning_authority, [])` would have given you what you were after.

@gnott the last argument for `dict.get()` and `getattr()` is a default value that is returned in the case that the api cannot return a value for the key/attribute passed.